### PR TITLE
CHE-3609: Create visual Git history window instead of old one

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add new visual Git history window instead of old one.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/3609

#### Changelog
Git history panel changed to a window to enable visualization of commit changes.

#### Release Notes
We've replaced the git history panel with a new window. This allows you to see the changes from each commit visually with a diff viewer. You can get to the git history from the git menu, or you can right-click on a file in the project explorer and select git history from the context menu.

The git history window will show you all the commits:

git_history_1

Double clicking on a revision item, or clicking the 'Compare' button will open the visual diff viewer:

git_history_2

### Tests written?
Yes

### CODE REVIEW

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java [hunk 1]
```diff
@@ -98,6 +101,25 @@ public final void writeTo(OutputStream out) throws IOException {
         }
     }
 
+    private List<DiffEntry> EmptyToCommit(String commitId, DiffFormatter formatter) throws IOException {
```
Comment (vzhukovs) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java]: `EmptyToCommit` method name should starts with lowercase

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java [hunk 2]
```diff
@@ -98,6 +101,25 @@ public final void writeTo(OutputStream out) throws IOException {
...
+        ObjectId commit = repository.resolve(commitId);
+        if (commit == null) {
```
Comment (vzhukovs) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java]: FYI, guava has utility method `com.google.common.base.Preconditions#checkArgument(boolean, java.lang.Object)`

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java [hunk 3]
```diff
@@ -10,449 +10,202 @@
...
-     *         if <code>true</code> compare with index, else - with working tree
-     */
-    private void doDiffWithNotCommitted(@NotNull List<String> filePatterns, @Nullable final Revision revision, final boolean isCached) {
-        if (revision == null) {
-            return;
-        }
-
-        service.diff(appContext.getDevMachine(), project.getLocation(), filePatterns, RAW, false, 0, revision.getId(), isCached)
-               .then(new Operation<String>() {
+    private void FetchRevisions() {
```
Comment (vzhukovs) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java]: Method name should starts with lowercase letter.

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java [hunk 4]
```diff
@@ -46,6 +45,7 @@
 @Singleton
 public class ResetToCommitPresenter implements ResetToCommitView.ActionDelegate {
     public static final String RESET_COMMAND_NAME = "Git reset to commit";
+    public static final String LOG_COMMAND_NAME = "Git log";
```
Comment (vzhukovs) [inline plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java]: Formatting.

**Comment by vzhukovs** [comment]
Other seems ok

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java (hunk 1 above)
Comment (vinokurig) [reply wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java]: fixed

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java (hunk 2 above)
Comment (vinokurig) [reply wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitDiffPage.java]: done

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java (hunk 3 above)
Comment (vinokurig) [reply plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/history/HistoryPresenter.java]: fixed

File: plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java (hunk 4 above)
Comment (vinokurig) [reply plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/reset/commit/ResetToCommitPresenter.java]: fixed

**Review by vzhukovs — APPROVED** [review]

**Review by vparfonov — APPROVED** [review]

**Pull request merged**